llvm/include/llvm/IR/IntrinsicsAArch64.td

```diff
@@ -76,6 +76,19 @@ def int_aarch64_prefetch : Intrinsic<[],
      ]>,
     ClangBuiltin<"__builtin_arm_prefetch">;
 
+def int_aarch64_range_prefetch_imm : Intrinsic<[],
+    [llvm_ptr_ty, llvm_i32_ty, llvm_i32_ty, llvm_i32_ty, llvm_i32_ty,
+     llvm_i32_ty, llvm_i64_ty],
+    [IntrInaccessibleMemOrArgMemOnly, IntrWillReturn, ReadOnly<ArgIndex<0>>,
+     ImmArg<ArgIndex<1>>, ImmArg<ArgIndex<2>>, ImmArg<ArgIndex<3>>, ImmArg<ArgIndex<4>>,
+     ImmArg<ArgIndex<5>>, ImmArg<ArgIndex<6>>]>;
```

Comment:
I'm not against having such, but do we need a user friendly LLVM intrinsic? You could let clang do the encoding and keep the IR side simple.